### PR TITLE
Move more parser tests to syntax tests

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by axic** [comment]
Left a couple of tests in `SolidityParser.cpp`, which had different errors (`TypeError` or `DeclarationError`), but did fix a couple of them during the transition.

File: test/libsolidity/syntaxTests/parsing/location_specifiers_for_params.sol [hunk 1]
```diff
@@ -0,0 +1,5 @@
+contract Foo {
+    function f(uint[] storage constant x, uint[] memory y) internal { }
```
Comment (axic) [inline test/libsolidity/syntaxTests/parsing/location_specifiers_for_params.sol:2]: `storage constant` is invalid (`Storage location has to be "memory" (or unspecified) for constants.`). Should we leave it as an error or use `constant` rather on the `memory` argument?

File: test/libsolidity/SolidityParser.cpp [hunk 2]
```diff
@@ -112,52 +112,6 @@ while(0)
...
-
-BOOST_AUTO_TEST_CASE(two_exact_functions)
-{
-	char const* text = R"(
-		contract test {
-			function fun(uint a) returns(uint r) { return a; }
-			function fun(uint a) returns(uint r) { return a; }
-		}
-	)";
-	// with support of overloaded functions, during parsing,
```
Comment (chriseth) [inline test/libsolidity/SolidityParser.cpp:155]: This should not be removed.

**Review by chriseth — DISMISSED** [review]
Fine apart from the missing comments. I did not check the tests in details nor did I count them.

File: test/libsolidity/syntaxTests/parsing/location_specifiers_for_params.sol (hunk 1 above)
Comment (chriseth) [reply test/libsolidity/syntaxTests/parsing/location_specifiers_for_params.sol:2]: I would leave it like it is.

File: test/libsolidity/SolidityParser.cpp (hunk 2 above)
Comment (axic) [reply test/libsolidity/SolidityParser.cpp:155]: I can add them back, though they are not relevant since syntax tests run the analysis phase.

Comment (chriseth) [reply test/libsolidity/SolidityParser.cpp:155]: ok right, I actually didn't read the comment, just saw that it was not retained :)

**Review by chriseth — APPROVED** [review]

**Comment by axic** [comment]
Added back the comment, but I think it is not needed there. I did manually verify the test cases (the proper ones removed and proper expectations, compared to the old version, but any extra eyes are welcome.

cc @bit-shift @leonardoalt

**Comment by erak** [comment]
@axic Looks good to me. I counted 52 tests that have been removed and 54 that have been added, so I guess there're all covered ;)
The only thing that caught my attention is that a lot of the tests have redundant expectations, e.g. ``No visibility specified. Defaulting to "public".`` I'm wondering if we should adjust the contracts under test such that the amount of redundant warnings is reduced and then explicitly checked in dedicated tests. But I think this can also be part of a separate discussion.

**Comment by chriseth** [comment]
I would say we should merge this first and then update the tests to 0.5.0 syntax, i.e. add public and stuff. Of course only if that is not the issue under test.

**Comment by axic** [comment]
> I would say we should merge this first and then update the tests to 0.5.0 syntax, i.e. add public and stuff. 

Yes, that would be the best approach, these are only moving test cases (mostly) verbatim, 0.5.0 fixes should be applied after.

**Pull request merged**